server/player/src/packet_handlers/inventory.rs

```diff
@@ -427,17 +424,16 @@ fn handle_double_click(
 
     let new_picked_count = {
         let mut current_count;
-        let item_type;
 
         // Get information about the currently picked item
-        if let Some(picked) = world.try_get_mut::<PickedItem>(player) {
-            stack_size = picked.0.ty.stack_size() as u8;
-            current_count = picked.0.amount;
-            item_type = picked.0.ty;
-        } else {
+        let picked = world.try_get_mut::<PickedItem>(player);
+        if picked.is_none() {
```

Comment:
Rather than unwrapping the option after the `is_none()` check, this should be changed to a match:

```rust
let picked = match picked {
    Some(picked) => picked.0,
    None => return Ok(()),
}
```